docs/aws/audit/dynamodbmonitoring/rules/ddb_table_backup_enabled.mdx

```diff
@@ -24,6 +24,208 @@ HIPAA, NIST, SOC2, ISO27001, HITRUST, AWSWAF, NISTCSF
+   - A confirmation dialog will appear. Confirm the action by clicking "Enable point-in-time recovery."
+
+By following these steps, you ensure that continuous backups are enabled for your DynamoDB tables, providing point-in-time recovery capabilities.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that DynamoDB tables have continuous backups enabled using AWS CLI, follow these steps:
+
+1. **List Existing DynamoDB Tables:**
+   First, identify the DynamoDB tables in your AWS account.
+   ```sh
+   aws dynamodb list-tables
+   ```
+
+2. **Describe Table to Check Backup Status:**
+   For each table, check if continuous backups (Point-in-Time Recovery) are enabled.
+   ```sh
+   aws dynamodb describe-continuous-backups --table-name <TableName>
+   ```
+
+3. **Enable Continuous Backups:**
+   If continuous backups are not enabled, enable them using the following command:
+   ```sh
+   aws dynamodb update-continuous-backups --table-name <TableName> --point-in-time-recovery-specification PointInTimeRecoveryEnabled=true
+   ```
+
+4. **Verify Backup Status:**
+   After enabling, verify that the continuous backups are now enabled.
+   ```sh
+   aws dynamodb describe-continuous-backups --table-name <TableName>
+   ```
+
+By following these steps, you can ensure that continuous backups are enabled for your DynamoDB tables using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that DynamoDB tables have continuous backups enabled using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by creating a `~/.aws/credentials` file.
+
+3. **Python Script to Enable Continuous Backups**:
+   Use the following Python script to enable continuous backups for all DynamoDB tables in your AWS account:
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon DynamoDB
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   # Initialize DynamoDB client
+   dynamodb = session.client('dynamodb')
+
+   # List all DynamoDB tables
+   tables = dynamodb.list_tables()['TableNames']
+
+   # Enable continuous backups for each table
+   for table in tables:
+       try:
+           response = dynamodb.update_continuous_backups(
+               TableName=table,
+               PointInTimeRecoverySpecification={
+                   'PointInTimeRecoveryEnabled': True
+               }
+           )
+           print(f"Continuous backups enabled for table: {table}")
+       except Exception as e:
+           print(f"Error enabling continuous backups for table {table}: {e}")
+   ```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>